src/object.cpp

```diff
@@ -1668,25 +1710,148 @@ int Object::ReorderByXPos(FunctorParams *functorParams)
     return FUNCTOR_CONTINUE;
 }
 
+bool Object::GenerateBoundingBox(int *ulx, int *uly, int *lrx, int *lry)
+{
+    // Set integers to extremes
+    *ulx = INT_MAX;
+    *uly = INT_MAX;
+    *lrx = INT_MIN;
+    *lry = INT_MIN;
+    ListOfObjects childrenWithFacsimileInterface;
+    InterfaceComparison ic(INTERFACE_FACSIMILE);
+    this->FindAllDescendantByComparison(&childrenWithFacsimileInterface, &ic);
+    bool result = false;
+    for (auto it = childrenWithFacsimileInterface.begin(); it != childrenWithFacsimileInterface.end(); ++it) {
+        FacsimileInterface *fi = dynamic_cast<FacsimileInterface *>(*it);
+        assert(fi);
+        if (!(*it)->Is(SYL) && fi->HasFacs()) {
+            Zone *zone = fi->GetZone();
+            assert(zone);
+            *ulx = std::min(*ulx, zone->GetUlx());
+            *uly = std::min(*uly, zone->GetUly());
+            *lrx = std::max(*lrx, zone->GetLrx());
+            *lry = std::max(*lry, zone->GetLry());
+            result |= true;
+        }
+    }
+    return result;
+}
+
 int Object::SetChildZones(FunctorParams *functorParams)
```

Comment:
This should definitely not have its main implementation in `Object` because it overloads it with something too specific. Maybe you can have a interface pseudo functor? See `TimeSpanningInterface::InterfaceResetDrawing` for example. Something like `FacsimileInterface::InterfaceSetChildZones` redirect from Object, or better, from LayerElement and Staff, which I think are the only two classes implementing FacsimilerInterface.